core/src/test/java/io/grpc/internal/BinaryLogProviderTest.java

```diff
@@ -16,20 +16,89 @@
 
 package io.grpc.internal;
 
+import static com.google.common.truth.Truth.assertThat;
+import static junit.framework.TestCase.assertEquals;
 import static org.junit.Assert.assertSame;
 
+import io.grpc.CallOptions;
 import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingServerCall;
+import io.grpc.ForwardingServerCallListener;
+import io.grpc.ManagedChannel;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
 import io.grpc.ReplacingClassLoader;
+import io.grpc.Server;
+import io.grpc.ServerCall;
+import io.grpc.ServerCall.Listener;
+import io.grpc.ServerCallHandler;
 import io.grpc.ServerInterceptor;
+import io.grpc.ServerServiceDefinition;
+import io.grpc.ServiceDescriptor;
+import io.grpc.StringMarshaller;
+import io.grpc.inprocess.InProcessChannelBuilder;
+import io.grpc.inprocess.InProcessServerBuilder;
+import io.grpc.inprocess.InternalInProcessServerBuilder;
+import io.grpc.stub.ClientCalls;
+import io.grpc.stub.ServerCalls;
+import io.grpc.stub.StreamObserver;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
 import javax.annotation.Nullable;
+import org.junit.After;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 /** Unit tests for {@link BinaryLogProvider}. */
 @RunWith(JUnit4.class)
 public class BinaryLogProviderTest {
+  private static final MethodDescriptor<Integer, Integer> INCREMENT_BY_ONE =
+      MethodDescriptor.<Integer, Integer>newBuilder()
+          .setType(MethodDescriptor.MethodType.UNARY)
+          .setFullMethodName("some/incrementbyone")
+          .setRequestMarshaller(new IntegerMarshaller())
+          .setResponseMarshaller(new IntegerMarshaller())
+          .build();
+
   private final String serviceFile = "META-INF/services/io.grpc.internal.BinaryLogProvider";
+  private final IntegerMarshaller serverReqMarshaller = new IntegerMarshaller();
+  private final IntegerMarshaller serverRespMarshaller = new IntegerMarshaller();
+  private final MethodDescriptor<Integer, Integer> serverMethod =
+      INCREMENT_BY_ONE.toBuilder(serverReqMarshaller, serverRespMarshaller).build();
+  private final ServerServiceDefinition service =
+      ServerServiceDefinition.builder(new ServiceDescriptor("some", serverMethod))
+          .addMethod(
+              serverMethod,
+              ServerCalls.asyncUnaryCall(
+                  new ServerCalls.UnaryMethod<Integer, Integer>() {
+                    @Override
+                    public void invoke(Integer request, StreamObserver<Integer> responseObserver) {
+                      responseObserver.onNext(request + 1);
+                      responseObserver.onCompleted();
+                    }
+                  }
+              ))
+          .build();
+  private final List<Object> req = new ArrayList<Object>();
```

Comment:
These four fields (this and below) are confusing, as they aren't used directly by any infrastructure while also being non-obvious just by looking at them. Since only one method is looking at them, could you move them into that test method for now?

Reply:
Refactored